### PR TITLE
[SP-5448] Add message language support

### PR DESCRIPTION
**Pull request by Suryakantj**
1. MessageLanguage feature is implemented.
2. Unit tests added for message language feature. 
3. UI test cases are yet to be written as we have to add message language support in MetaApp

### CODE REVIEW

**Comment by carmelo-iriti** [comment]
Hi @Suryakantj, before reviewing this pr, can you please rebase on develop?

**Comment by Suryakantj** [comment]
> Hi @Suryakantj, before reviewing this pr, can you please rebase on develop?

@carmelo-iriti Yes , Just resolved the conflicts. Now PR is ready for review.

File: metaapp/src/androidTest/java/com/sourcepointmeta/metaapp/MessageLanguageTests.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.action.ViewActions.replaceText;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+
+@RunWith(AndroidJUnit4ClassRunner.class)
+public class MessageLanguageTests extends Utility {
```
[the remark this answers is not in the document]
Comment (Suryakantj) [reply metaapp/src/androidTest/java/com/sourcepointmeta/metaapp/MessageLanguageTests.java]: Yes converted java tests to kotlin as in MetaAppTestK.kt

File: metaapp/src/test/java/com/sourcepointmeta/metaapp/repository/PropertyListRepositoryTest.java [hunk 2]
```diff
@@ -88,11 +88,11 @@ public void updatePropertyTest() {
         Property property = StaticTestData.PROPERTIES.get(0);
         property.setId(1);
 
-        propertyListDao.update(property.getAccountID(), property.getPropertyID() , property.getProperty(), property.getPmID() , property.isStaging(), property.isNative(), property.getAuthId(), property.getId());
-        verify(propertyListDao).update(property.getAccountID(), property.getPropertyID(), property.getProperty(), property.getPmID(), property.isStaging(), property.isNative(), property.getAuthId(), property.getId());
+        propertyListDao.update(property.getAccountID(), property.getPropertyID() , property.getProperty(), property.getPmID() , property.isStaging(), property.isNative(), property.getAuthId(), property.getMessageLanguage(), property.getId());
+        verify(propertyListDao).update(property.getAccountID(), property.getPropertyID(), property.getProperty(), property.getPmID(), property.isStaging(), property.isNative(), property.getAuthId(), property.getMessageLanguage(), property.getId());
```
[the remark this answers is not in the document]
Comment (Suryakantj) [reply metaapp/src/test/java/com/sourcepointmeta/metaapp/repository/PropertyListRepositoryTest.java]: Here intention is to test update method of PropertyListDao class.

File: cmplibrary/src/test/java/com/sourcepoint/gdpr_cmplibrary/ConsentLibBuilderTest.java [hunk 3]
```diff
@@ -162,6 +162,12 @@ public void setAuthId() {
         assertEquals("authId", defaultBuilder.authId);
     }
 
+    @Test
```
[the remark this answers is not in the document]
Comment (Suryakantj) [reply cmplibrary/src/test/java/com/sourcepoint/gdpr_cmplibrary/ConsentLibBuilderTest.java:165]: Yes added check for entire enum

**Review by carmelo-iriti — COMMENTED** [review]
The intention is right, I don't understand how we can test the concrete logic of a mock object

File: metaapp/src/test/java/com/sourcepointmeta/metaapp/repository/PropertyListRepositoryTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (carmelo-iriti) [reply metaapp/src/test/java/com/sourcepointmeta/metaapp/repository/PropertyListRepositoryTest.java]: The intention is right, I don't understand how we can test the concrete logic of a mock object

**Comment by carmelo-iriti** [comment]
Please keep the tests for this feature in the original class you created ```MessageLanguageTests```

File: samples/metaapp/src/androidTest/java/com/sourcepointmeta/metaapp/MetaAppTestCases.kt [hunk 4]
```diff
@@ -116,6 +124,15 @@ class MetaAppTestCases {
             insertTextByResId(propId = R.id.etPMId, text = pmId)
         }
 
+        fun addMessageLanguagePropertyDetails() {
```
Comment (carmelo-iriti) [inline samples/metaapp/src/androidTest/java/com/sourcepointmeta/metaapp/MetaAppTestCases.kt:127]: @Suryakantj it looks like that the data for tests are missing

File: samples/metaapp/src/main/res/layout/activity_new_property.xml [hunk 5]
```diff
@@ -128,6 +128,18 @@
                 android:textSize="@dimen/text_size_20" />
         </android.support.design.widget.TextInputLayout>
 
+        <Spinner
+            android:id="@+id/spinnerMessageLanguage"
```
[the remark this answers is not in the document]
Comment (Suryakantj) [reply samples/metaapp/src/main/res/layout/activity_new_property.xml]: Done

**Review by carmelo-iriti — APPROVED** [review]

File: metaapp/src/test/java/com/sourcepointmeta/metaapp/repository/PropertyListRepositoryTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (carmelo-iriti) [reply metaapp/src/test/java/com/sourcepointmeta/metaapp/repository/PropertyListRepositoryTest.java]: We need to remove in the future this useless test

**Pull request merged**